package/src/contexts/attachmentPickerContext/AttachmentPickerContext.tsx

```diff
@@ -118,8 +121,19 @@ export const AttachmentPickerProvider = ({
   );
 };
 
-export const useAttachmentPickerContext = () =>
-  useContext(AttachmentPickerContext) as unknown as AttachmentPickerContextValue;
+export const useAttachmentPickerContext = () => {
+  const contextValue = useContext(
+    AttachmentPickerContext,
+  ) as unknown as AttachmentPickerContextValue;
+
+  if (contextValue === DEFAULT_BASE_CONTEXT_VALUE && !isTestEnvironment()) {
+    throw new Error(
+      `The useAttachmentPickerContext hook was called outside the AttachmentPickerContext provider. Make sure you have configured OverlayProvider component correctly - https://getstream.io/chat/docs/sdk/reactnative/basics/hello_stream_chat/#overlay-provider`,
+    );
+  }
```

Comment:
This way we only throw error if the components/hooks are being used in actual application and not in jest rendering.

If we allow error in for test suit as well, that opens up the pandora's box with fixing the tests by wrapping every component rendering with `OverlayProvider` and `Chat` and `Channel`. I think the tests that we have now are fine as it is. There is no amazing benefit from adding these wrappers everywhere.

Reply:
Implementation of isTestEnvironment is as following:

```ts
export const isTestEnvironment = () => process.env.NODE_ENV === 'test';
```

I am taking this reference for NODE_ENV from react-native's core codebase itself. So I think its safe to use it.